src/Parsers/ParserPartition.cpp

```diff
@@ -28,6 +30,14 @@ bool ParserPartition::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
         partition->id = partition_id->as<ASTLiteral &>().value.get<String>();
     }
+    else if (s_all.ignore(pos, expected))
+    {
+        ASTPtr value = makeASTFunction("tuple");
+        partition->value = value;
+        partition->children.push_back(value);
+        partition->fields_count = 0;
```

Comment:
i have considered the  table which has partition key, it will search all partition actually. ALL just interpret as tuple(),but not equal to tuple(),please refer to my example in pr.

Reply:
Maybe i can fix ASTPartition::fromatImpl just like
```
@@ -31,6 +31,10 @@ void ASTPartition::formatImpl(const FormatSettings & settings, FormatState & sta
 {
     if (value)
     {
+        if (all)
+        {
+            settings.ostr << "ALL";
+        }
         value->formatImpl(settings, state, frame);
     } 
```